js/js.libraries/src/core/collections.kt

```diff
@@ -85,13 +85,34 @@ public fun <K, V> mapOf(pair: Pair<K, V>): Map<K, V> = hashMapOf(pair)
  * Each element in the list gets replaced with the [value].
  */
 @SinceKotlin("1.2")
-public fun <T> MutableList<T>.fill(value: T) {
-    for (index in 0..lastIndex) {
+public fun <T> MutableList<T>.fill(value: T): Unit {
+    for (index in indices) {
         this[index] = value
     }
 }
 
 /**
+ * Randomly shuffles elements in this list.
+ *
+ * See: https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm
+ */
+@SinceKotlin("1.2")
+public fun <T> MutableList<T>.shuffle(): Unit {
+    (lastIndex downTo 1).forEach { i ->
+        rand(i + 1).let { j ->
+            swap(i, j)
```

Comment:
There's no `swap` in JS currently

Reply:
Fixed with a private method instead, thanks!